test/components/selectionBoxLayerTests.ts

```diff
@@ -0,0 +1,70 @@
+///<reference path="../testReference.ts" />
+
+var assert = chai.assert;
+
+describe("SelectionBoxLayer", () => {
+  it("boxVisible()", () => {
+    var svg = generateSVG();
+    var sbl = new Plottable.Component.SelectionBoxLayer();
+    sbl.renderTo(svg);
+
+    var selectionBox = svg.select(".selection-box");
+    assert.isTrue(selectionBox.empty(), "initilizes without box in DOM");
+
+    sbl.boxVisible(true);
+    selectionBox = svg.select(".selection-box");
+    assert.isFalse(selectionBox.empty(), "box is inserted in DOM when showing");
+
+    sbl.boxVisible(false);
+    selectionBox = svg.select(".selection-box");
+    assert.isTrue(selectionBox.empty(), "box is removed from DOM when not showing");
+
+    svg.remove();
+  });
+
+  it("bounds()", () => {
+    var svg = generateSVG();
+    var sbl = new Plottable.Component.SelectionBoxLayer();
+
+    var topLeft: Plottable.Point = {
+      x: 100,
+      y: 100
+    };
+    var bottomRight: Plottable.Point = {
+      x: 300,
+      y: 300
+    };
+    assert.doesNotThrow(() => sbl.bounds({
+      topLeft: topLeft,
+      bottomRight: bottomRight
+    }), Error, "can set bounds before anchoring");
+
+    sbl.boxVisible(true);
+    sbl.renderTo(svg);
+
+    function assertCorrectRendering(expectedTL: Plottable.Point, expectedBR: Plottable.Point, msg: string) {
+      var selectionBox = svg.select(".selection-box");
+      var bbox = (<SVGLocatable> <any> selectionBox.node()).getBBox();
```

Comment:
No, you can't cast `Element` (the return type of `node()`) directly to `SVGLocatable`.